pinecone/core/client/rest.py

```diff
@@ -1,59 +1,49 @@
-# coding: utf-8
-
 """
     Pinecone API
 
-    No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
+    No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)  # noqa: E501
 
     The version of the OpenAPI document: version not set
     Contact: support@pinecone.io
-    Generated by OpenAPI Generator (https://openapi-generator.tech)
-
-    Do not edit the class manually.
-"""  # noqa: E501
+    Generated by: https://openapi-generator.tech
+"""
 
 
 import io
 import json
 import logging
 import re
 import ssl
+from urllib.parse import urlencode
 
-from urllib.parse import urlencode, quote_plus
 import urllib3
 
-from pinecone.core.client.exceptions import (
-    ApiException,
-    UnauthorizedException,
-    ForbiddenException,
-    NotFoundException,
-    ServiceException,
-    ApiValueError,
-    BadRequestException,
-)
+from pinecone.core.client.exceptions import ApiException, UnauthorizedException, ForbiddenException, NotFoundException, ServiceException, ApiValueError
 
 
 logger = logging.getLogger(__name__)
 
 
 class RESTResponse(io.IOBase):
-    def __init__(self, resp) -> None:
+
+    def __init__(self, resp):
         self.urllib3_response = resp
         self.status = resp.status
         self.reason = resp.reason
         self.data = resp.data
 
     def getheaders(self):
         """Returns a dictionary of the response headers."""
-        return self.urllib3_response.headers
+        return self.urllib3_response.getheaders()
 
     def getheader(self, name, default=None):
         """Returns a given response header."""
         return self.urllib3_response.headers.get(name, default)
```

Comment:
This `return self.urllib3_response.headers.get(name, default)` is not showing as a change because it was handled correctly in the 7.0.1 generator that we are downgrading from. But the fact it is not changed means my template override was successfully used; without this template override, openapi 5.2.8 generates `return self.urllib3_response.getheader(name, default)` which causes a large amount of runtime deprecation notices to appear.